addons/l10n_pt_pos/models/pos_order.py

```diff
@@ -0,0 +1,220 @@
+import re
+
+from odoo.addons.l10n_pt_account.utils.hashing import L10nPtHashingUtils
+from odoo import models, fields, _, api
+from odoo.exceptions import UserError
+from odoo.tools import float_repr, format_date
+
+
+class PosOrder(models.Model):
+    _inherit = "pos.order"
+
+    country_code = fields.Char(related='company_id.account_fiscal_country_id.code', depends=['company_id.account_fiscal_country_id'])
+    l10n_pt_pos_document_number = fields.Char(string='Portuguese Document Number', compute='_compute_l10n_pt_pos_document_number', store=True)
+    l10n_pt_pos_inalterable_hash = fields.Char(string="Inalterability Hash", readonly=True, copy=False)
+    l10n_pt_pos_qr_code_str = fields.Char(string='Portuguese QR Code', compute='_compute_l10n_pt_pos_qr_code_str', store=True)
+    l10n_pt_pos_inalterable_hash_short = fields.Char(string='Short version of the Portuguese hash', compute='_compute_l10n_pt_pos_inalterable_hash_short')
+    l10n_pt_pos_atcud = fields.Char(string='Portuguese ATCUD', compute='_compute_l10n_pt_pos_atcud', store=True)
+
+    @api.depends('name', 'company_id.account_fiscal_country_id.code')
+    def _compute_l10n_pt_pos_document_number(self):
+        for order in self.filtered(lambda o: (
+            o.company_id.account_fiscal_country_id.code == 'PT'
+            and o.config_id.l10n_pt_pos_official_series_id
+            and o.name
+            and not o.l10n_pt_pos_document_number
+        )):
+            sequence_prefix, sequence_number = order._l10n_pt_pos_get_sequence_info()
+            order.l10n_pt_pos_document_number = f"pos_order {sequence_prefix}/{sequence_number}"
+
+    @api.depends('l10n_pt_pos_inalterable_hash')
+    def _compute_l10n_pt_pos_inalterable_hash_short(self):
+        for order in self:
+            order.l10n_pt_pos_inalterable_hash_short = L10nPtHashingUtils._l10n_pt_get_short_hash(order.l10n_pt_pos_inalterable_hash)
+
+    @api.depends('name', 'config_id.l10n_pt_pos_official_series_id.code', 'l10n_pt_pos_inalterable_hash')
+    def _compute_l10n_pt_pos_atcud(self):
+        for order in self:
+            if (
+                order.company_id.account_fiscal_country_id.code == 'PT'
+                and order.config_id.l10n_pt_pos_official_series_id
+                and order.l10n_pt_pos_inalterable_hash
+                and not order.l10n_pt_pos_atcud
```

Comment:
If the `l10n_pt_pos_atcud` is already set, this will be `False` and you will then execute the `else` block :boom: